core/textile/utils.go

```diff
@@ -23,3 +28,43 @@ func parseDbIDFromString(dbID string) (*thread.ID, error) {
 
 	return &id, nil
 }
+
+type deterministicThreadVariant string
+
+var (
+	metathreadThreadVariant deterministicThreadVariant = "metathread"
+)
+
+func newDeterministicThreadID(st *store.Store, threadVariant deterministicThreadVariant) (thread.ID, error) {
```

Comment:
since store is already an interface, you probably don't need to pass it as a pointer again (because then it would be a double pointer to the actual store implementation)

```suggestion
func newDeterministicThreadID(st store.Store, threadVariant deterministicThreadVariant) (thread.ID, error) {
```

Reply:
Fixed in 4944baf